app/src/main/java/com/golabiusz/snake/Snake.java

```diff
@@ -1,4 +1,242 @@
 package com.golabiusz.snake;
 
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Matrix;
+import android.graphics.Paint;
+import android.graphics.Point;
+import org.jetbrains.annotations.NotNull;
+import java.util.ArrayList;
+
 public class Snake {
+  private enum Heading {
+    UP,
+    RIGHT,
+    DOWN,
+    LEFT
+  }
+
+  private final Heading STARTING_HEADING = Heading.RIGHT;
+
+  private Heading heading = STARTING_HEADING;
+
+  // The location in the grid of all the segments
+  private final ArrayList<Point> segmentLocations;
+
+  private final int segmentSize;
+
+  private Bitmap bitmapHeadRight;
+  private Bitmap bitmapHeadLeft;
+  private Bitmap bitmapHeadUp;
+  private Bitmap bitmapHeadDown;
+
+  private Bitmap bitmapBody;
+
+  Snake(Context context, int segmentSize) {
+    segmentLocations = new ArrayList<>();
```

Comment:
Will this work, or should be `... = new ArrayList<Point>()`;

Reply:
It works, but good point :+1:

Reply:
Well... As I checked and sent you, there is no need to explicit type during initialization.